dali/test/python/conditionals/test_pipeline_conditionals.py

```diff
@@ -617,3 +617,135 @@ def one_return():
                                 " (both `if` branches). The `else` branch must also have a return"
                                 " statement.")):
         one_return()
+
+
+def _tensor_arg_permute_batch_params():
+    batch_sizes = [1, 5, 8]
+    inp0 = [[np.full((2, 2), i, dtype=np.float32) for i in range(batch_size)]
+            for batch_size in batch_sizes]
+    mask_batches = [
+        np.array([i % 2 for i in range(batch_size)], dtype=bool) for batch_size in batch_sizes
+    ]
+    kwarg_batches = [np.array([pred for pred in mask], dtype=np.int32) for mask in mask_batches]
+    return (inp0, ), mask_batches, {'indices': kwarg_batches}
+
+
+def _tensor_arg_transform_per_dim_params(arg_name):
+
+    def inner():
+        batch_sizes = [5, 1, 2, 8]
+        mask_batches = [
+            np.array([i % 2 for i in range(batch_size)], dtype=bool) for batch_size in batch_sizes
+        ]
+        kwarg_batches = [
+            np.array([[pred, pred] for pred in mask], dtype=np.float32) for mask in mask_batches
+        ]
+        return tuple(), mask_batches, {arg_name: kwarg_batches}
+
+    return inner
+
+
+def _tensor_arg_rotate_params():
+    batch_sizes = [3, 1, 2, 4]
+    mask_batches = [
+        np.array([i % 2 for i in range(batch_size)], dtype=bool) for batch_size in batch_sizes
+    ]
+    kwarg_batches = [
+        np.array([10 + 45 * pred for pred in mask], dtype=np.float32) for mask in mask_batches
+    ]
+    return tuple(), mask_batches, {'angle': kwarg_batches}
+
+
+def _tensor_arg_roi_random_crop_params():
+    batch_sizes = [1, 2, 7, 3]
+    crop_shape = [[
+        np.array([100 * i + 50, 200 * i + 50, 3], dtype=np.int32) for i in range(batch_size)
+    ] for batch_size in batch_sizes]
+    roi_start = [[
+        np.array([sample[0] // 2, sample[1] // 2, sample[2]], dtype=np.int32) for sample in batch
+    ] for batch in crop_shape]
+    mask_batches = [
+        np.array([i % 2 for i in range(batch_size)], dtype=bool) for batch_size in batch_sizes
+    ]
+    return tuple(), mask_batches, {
+        'crop_shape': crop_shape,
+        'roi_start': roi_start,
+        'roi_end': crop_shape
+    }
+
+
+def _tensor_arg_shape_kwarg():
+    batch_sizes = [1, 2, 3, 16, 5]
+    shape = [[np.array([1 + 3 * i, 2 * (i + 1) - 1], dtype=np.int32) for i in range(batch_size)]
+             for batch_size in batch_sizes]
+    mask_batches = [
+        np.array([i % 2 for i in range(batch_size)], dtype=bool) for batch_size in batch_sizes
+    ]
+    return tuple(), mask_batches, {'shape': shape}
+
+
+# Test operators that infer their batch sizes from the tensor argument inputs
+@params(fn.permute_batch, fn.roi_random_crop, fn.transforms.crop, fn.transforms.scale,
+        fn.transforms.shear, fn.transforms.translation, fn.transforms.rotation,
+        fn.random.uniform, fn.random.normal, fn.random.coin_flip)
+def test_named_tensor_arguments(op):
+
+    ops2params = {
+        fn.permute_batch: _tensor_arg_permute_batch_params,
```

Comment:
This one is a nasty little bugger - in principle, the output batch_size comes from the `indices` argument and not the input.